core/cache/fieldwatcher.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+/*
+ * CODE GENERATED BY juju/scripts/generate-watchers/main.go.
```

Comment:
I see there is a _juju/generate_ package. This should probably live there if it sticks.